xarray/core/common.py

```diff
@@ -474,7 +475,7 @@ def _calc_assign_results(
 
     def assign_coords(
         self,
-        coords: Mapping[Any, Any] | None = None,
+        coords: Mapping[Any, Any] | Coordinates | None = None,
```

Comment:
```suggestion
        coords: Mapping | None = None,
```

`Coordinates` is itself a Mapping so it is redundant to add it here (but still useful to mention it in the docstrings). I think `Mapping` is equivalent to `Mapping[Any, Any]`.

Reply:
Ah great point, thanks

Reply:
I think it is good to be explicit.
E.g. we always annotate e.g. `Hashable | None` even though None is hashable.
It is just another hint for the user and type checkers usually don't care.

Also now with these changes this PR basically only changes the docstring, haha.

Reply:
@headtr1ck is that a case for adding `Coordinates` or having `Mapping[Any, Any]`?

I can kinda see the point of having `Coordinates`, though it's also now in the docstring. And maybe good to educate folks that it is a mapping?

I do agree with @benbovy re `Mapping[Any, Any]` — that seems to add no value — it's not accurate that we can pass `Any`, we just haven't written it out.

Reply:
I assume @headtr1ck was referring to adding Coordinates.

I agree with both point of views (I don't have strong opinions).

The case `Hashable | None` seems more sensible to me as these two usually have very different semantics. For `Mapping | Coordinates` less so (a slight difference: a Coordinates object may also pass indexes).